lib/module.js

```diff
@@ -163,25 +163,28 @@ export default function SentryModule (moduleOptions) {
   }
 
   // Enable publishing of sourcemaps
-  if (!options.disabled) {
-    this.extendBuild((config, { isClient, isModern, isDev }) => {
-      if (!options.publishRelease || isDev) {
-        return
-      }
-
-      config.devtool = options.sourceMapStyle
-
-      // when not in spa mode upload only at server build
-      if (isClient && this.options.mode !== 'spa') {
-        return
-      }
-      // when in spa mode upload only at modern build if enabled
-      if (isClient && !isModern && this.options.modern) {
-        return
-      }
-
-      config.plugins.push(new WebpackPlugin(options.webpackConfig))
-      logger.info('Enabling uploading of release sourcemaps to Sentry')
+  if (options.publishRelease && !options.disabled && !this.options.dev) {
+    this.nuxt.hook('webpack:config', (webpackConfigs) => {
+      webpackConfigs.forEach((config) => {
+        config.devtool = options.sourceMapStyle
+
+        const isClient = config.name === 'modern' || config.name === 'client'
+        const isModern = config.name === 'modern'
+
+        // when not in spa/universal mode upload only at server build
+        if (isClient && this.options.mode !== 'spa' && this.options.mode !== 'universal') {
```

Comment:
Why was `this.options.mode !== 'universal'` condition added here?
I'd say we want to return here for client config in universal mode.

Reply:
Humm, I realized with the change in the log that the module was not uploading my client files to sentry, which is why I added this change. Why woudn't we want to send map files for client build to sentry ?

Reply:
Nuxt builds all configs sequentially. First `client` then `modern` then `server`. We only want to add the sentry webpack plugin on the `server` stage since at that point client and server code is built. We don't want to upload when building clients as that would be overwritten later on server stage anyway.

Of course, when building SPA-only, we need to upload on client stage already as there will be no server to build.

Reply:
Ping

Reply:
Are you sure that the WebpackPlugin will upload files that were not processed by the plugin ? this seems kinda odd to me. Anyway, if that is the case, I would change the code to directly add the plugin to the last config (without assuming any build order):

```js
webpackConfigs.forEach((config) => {
  config.devtool = options.sourceMapStyle
})

const config = webpackConfigs[webpackConfigs.length - 1]

config.plugins = config.plugins || []
config.plugins.push(new WebpackPlugin(options.webpackConfig))

logger.info(`Enabling uploading of release sourcemaps to Sentry for "${config.name}" build`)
```

Reply:
> Are you sure that the WebpackPlugin will upload files that were not processed by the plugin ?

Plugin is told to upload stuff from `.nuxt/dist/server` and `.nuxt/dist/client`. On client building, it would not have server bundle and maps for it yet, so we postpone it until the last stage.

> Anyway, if that is the case, I would change the code to directly add the plugin to the last config (without assuming any build order):

That sounds good as long as we can rely on order of configs matching build order.

Reply:
I think it's a weird usage for a Webpack plugin linked to a build to manipulate code from another build. Especially since the `options.build.parallel` option might break any assumption we make on the order of execution anyway. See [here](https://github.com/nuxt/nuxt.js/blob/903c3de5b65edb1c3eed763371a72c1b534e22f7/packages/webpack/src/builder.js#L100-L102).

As I see it, either you use add `SentryWebpackPlugin` for every webpack config, or you use the `build:compile` hook to detect when the build is over, and use `@sentry/cli` directly to upload the release.

```js
const configNames = []
this.nuxt.hook('webpack:config', (webpackConfigs) => {
      webpackConfigs.forEach((config) => {
        config.devtool = options.sourceMapStyle
        configNames.push(config.name)
      })
})

this.nuxt.hook('build:compiled', ({ name, compiler, stats }) => {
  const idx = configNames.indexOf(name)
  if (idx !== -1 && configNames.splice(idx, 1)  && configNames.length === 0) {
    // Publish the release using @sentry/cli
  }
})
```

Reply:
Or you prevent parallel build:
```js
    if (options.build.parallel) {
      throw new Error('@nuxtjs/sentry is not able to publish a release in parallel build mode.')
    }
```

Reply:
I'm certainly open to investigating a better solution. Either by you or myself when I have time.

Using `cli` might be a solution but probably requires extra dependency (how big?) and would be a bit awkward to run from the code maybe (unless it has programmatic interface also)?

Reply:
Actually i am wrong, parallel build is only for dev, for which publish is disabled...

Reply:
https://www.npmjs.com/package/@sentry/webpack-plugin?activeTab=dependencies is actually a dependency of webpack plugin

Reply:
Anyway, you can keep this in mind as an improvement to this module's code base. My latest commit to this PR is not really affected by this.